src/Color/Scale.purs

```diff
@@ -75,15 +93,29 @@ uniformScale mode b middle e = colorScale mode b stops e
 
 -- | Add a stop to a color scale.
 addStop :: ColorScale -> Color -> Number -> ColorScale
-addStop (ColorScale mode b middle e) c r =
-  ColorScale mode b (insertBy (comparing stopRatio) stop middle) e
+addStop (ColorScale mode b) c r = ColorScale mode $ addStop' b c r
+
+addStop' :: ColorScaleBuilder -> Color -> Number -> ColorScaleBuilder
+addStop' (ColorScaleBuilder b middle e) c r =
+  ColorScaleBuilder b (insertBy (comparing stopRatio) stop middle) e
     where stop = colorStop c r
 
+sample :: Sampler ColorScale
+sample (ColorScale mode scale) = mkSimpleSampler (mix mode) scale
+
+type Sampler a = a -> Number -> Color
```

Comment:
IMHO: this type synonym just adds mental overhead when reading the type definitions below.